docs/tutorial.md

```diff
@@ -0,0 +1,274 @@
+# Tutorial
+
+**Table of Contents**
+
+- [Basic](#advance)
+  - [Creating a bot](#creating-a-bot)
+  - [Listening for an event](#listening-for-an-event)
+    - [Echo bot](#echo-bot)
+    - [Welcome Bot](#welcome-bot)
+  - [Callbacks](#callbacks)
+    - [Craft Stick Bot](#craft-stick-bot)
+    - [Consuming Bot](#consuming-bot)
+    - [Fisherman Bot](#fisherman-bot)
+- [FAQ](#faq)
+  - [How to create discord bot with mineflayer](#how-to-create-discord-bot-with-mineflayer)
+  - [How to run bot on android](#how-to-run-bot-on-android)
+
+## Basic
+
+For the following, we assume Mineflayer is installed and you have a text editor.
+
+### Creating a Bot
+
+To get a feeling for the Mineflayer and how to use it, please open a text editor and follow the next few steps.
+
+First, create an instance of the Bot.
+
+```js
+var mineflayer = require("mineflayer");
+var bot = mineflayer.createBot({
+  host: "localhost", // optional / minecraft server address
+  port: 25565, // optional
+  username: "email@example.com", // email and password are required only for
+  password: "12345678", // online-mode=true servers
+  version: false, // false corresponds to auto version detection (that's the default), put for example "1.8.8" if you need a specific version
+});
+```
+
+Say hi to the chat after you spawn in, using [`spawn event`](http://mineflayer.prismarine.js.org/#/api?id=quotspawnquot) and [`chat method`](http://mineflayer.prismarine.js.org/#/api?id=botchatmessage).
+
+```js
+// ...
+
+bot.once("spawn", function () {
+  bot.chat("hi!");
+  // `chat` method for sending message or commad to the chat.
+});
+```
+
+For full methods list see the documentation at [API-Bot-methods](http://mineflayer.prismarine.js.org/#/api?id=methods)
+
+### Listening for an event
+
+Bot object has many useful built in [`events`](http://mineflayer.prismarine.js.org/#/api?id=events).
+You can listen for an event by using either `on` method or `once` method of the Bot object, wich takes the name of event and a function.
```

Comment:
Describe a difference between `bot.on()` and `bot.once()`
Use `bot.<method>()` notation